src/js/graphql_endpoint/modules/schema_to_graphql.ts

```diff
@@ -0,0 +1,168 @@
+import {
+    GraphQLBoolean,
+    GraphQLFieldConfigMap,
+    GraphQLInt,
+    GraphQLList,
+    GraphQLObjectType,
+    GraphQLOutputType,
+    GraphQLString,
+    GraphQLUnionType,
+} from "graphql";
+import { RawNode } from "./dgraph_client";
+import { Schema, SchemaProperty } from "./schema_client";
+import { RiskType } from "./schema";
+
+export type GqlTypeMap = Map<string, GraphQLObjectType>;
+
+function propertyToGraphql(
+    property: SchemaProperty,
+    typeMap: GqlTypeMap
+): GraphQLOutputType {
+    if (typeMap.size == 0) {
+        throw new Error(
+            "We expect to only execute this function once the ResolutionMap is full"
+        );
+    }
+
+    const hardcodedOverride = getHardcodedOverride(property, typeMap);
+    if (hardcodedOverride) {
+        return hardcodedOverride;
+    }
+
+    let prim: GraphQLOutputType = undefined;
+    if (property.name == "uid") {
+        // Special case
+        return GraphQLInt;
+    }
+
+    if (property.primitive == "Int") {
+        prim = GraphQLInt;
+    } else if (property.primitive == "Bool") {
+        prim = GraphQLBoolean;
+    } else if (property.primitive == "Str") {
+        prim = GraphQLString;
+    } else {
+        if (typeMap.has(property.primitive)) {
+            prim = typeMap.get(property.primitive);
+        } else {
+            throw new Error(
+                `Couldn't resolve property ${property.name}: ${property.primitive}`
+            );
+        }
+    }
+
+    if (property.is_set) {
+        return GraphQLList(prim);
+    } else {
+        return prim;
+    }
+}
+
+function getHardcodedOverride(
+    property: SchemaProperty,
+    typeMap: GqlTypeMap
+): GraphQLOutputType | undefined {
+    /* grapl_analyzerlib is *wrong*, and fixing it breaks other services.
+       so, for now, just manually override and fix it in a followup PR. 
+       context at:
+       https://grapl-internal.slack.com/archives/C017PLQ8TCZ/p1617991501057700
```

Comment:
Sounds like we should have an issue in the tracker for this.

Reply:
done https://github.com/grapl-security/issue-tracker/issues/388